src/main/frontend/components/diff.cljs

```diff
@@ -66,75 +67,82 @@
-                                    {:commit? false
-                                     :re-render-root? true})
-                   (mark-as-resolved path))
-                 :background "green")
-
-               [:span.pl-2.pr-2 "or"]
-
-               (ui/button "Keep local"
-                 :on-click
-                 (fn []
-                   ;; overwrite the file
-                   (swap! state/state
-                          assoc-in [:github/contents repo remote-oid path] local-content)
-                   (mark-as-resolved path))
-                 :background "pink")
-
-               [:span.pl-2.pr-2 "or"]
-
-               (ui/button "Edit"
-                 :on-click
-                 (fn []
-                   (reset! *edit? true)))]
-
-              :else
-              nil)])))]))
+     (let [content (get contents path)]
+       (if (or (and delete? (nil? content))
+               content)
+         (let [local-content (db/get-file path)]
+           (if (not= content local-content)
+             (let [local-content (or local-content "")
+                   content (or content "")
+                   diff (medley/indexed (diff/diff local-content content))
+                   diff? (some (fn [[_idx {:keys [added removed]}]]
+                                 (or added removed))
+                               diff)]
+               [:div.pre-line-white-space.p-2 {:class (if collapse? "hidden")
+                                               :style {:overflow "auto"}}
+                (if edit?
+                  [:div.grid.grid-cols-2.gap-1
+                   (diff-cp diff)
+                   (ui/textarea
+                    {:default-value local-content
+                     :on-change (fn [e]
+                                  (reset! *edit-content (util/evalue e)))})]
+                  (diff-cp diff))
+
+                (cond
+                  edit?
+                  [:div.mt-2
+                   (ui/button "Save"
+                              :on-click
+                              (fn []
+                                (reset! *edit? false)
+                                (let [new-content @*edit-content]
+                                  (file/alter-file repo path new-content
+                                                   {:commit? false
+                                                    :re-render-root? true})
+                                  (swap! state/state
+                                         assoc-in [:github/contents repo remote-oid path] new-content)
+                                  (mark-as-resolved path))))]
+
+                  diff?
+                  [:div.mt-2
+                   (ui/button "Use remote"
+                              :on-click
+                              (fn []
+                                ;; overwrite the file
+                                (if delete?
+                                  (file/remove-file! repo path)
```

Comment:
`remove-file!` calls `git/remove-file` and `fs/unlink`, but these should be done by `git merge`?

Reply:
I'm not quite sure about this, because the reason it's in the diff page because git merge failed, should we call `git merge` again?

Reply:
Oh, I misunderstood. But why does the current code work?

Reply:
The reason is that if the remote delete a file, logseq will treat it as an empty file. The file is not deleted but empty.